generic/tests/ping.py

```diff
@@ -58,76 +64,141 @@ def _get_loss_ratio(output):
+
+            error_context.context("Ping test on nic %s (index %d) from host"
+                                  " side" % (nic_name, i))
+            for size in packet_sizes:
+                for interval_t in interval_times:
+                    logging.info("Ping with packet size %s" % size)
+                    status, output = utils_test.ping(ip, 30,
+                                                     interval=interval_t,
+                                                     packetsize=size,
+                                                     interface=interface,
+                                                     timeout=60)
+                    _get_loss_ratio(output)
+
+                    if status != 0:
+                        test.fail("Ping failed, status: %s,"
+                                  " output: %s" % (status, output))
+
+            # ping with exceeded packages
+            error_context.context("Ping test with exceeded package",
+                                  logging.info)
+            status, output = utils_test.ping(ip, 30, packetsize=65508,
+                                             interface=interface, timeout=60)
+            result_str = "packet size 65508 is too large. Maximum is 65507"
+            if output.find(result_str) == -1:
+                test.fail(("Ping test with exceeded package failed,"
+                           " expected string: '%s' not found" %
+                           result_str))
+            # ping with flood
+            error_context.context("Flood ping test from host",
+                                  logging.info)
+            utils_test.ping(ip, None, flood=True, output_func=None,
+                            interface=interface, timeout=flood_minutes * 60)
+
+            error_context.context("Ping test after flood ping,"
+                                  " Check if the network is still alive",
+                                  logging.info)
+            status, output = utils_test.ping(ip, counts, interface=interface,
+                                             timeout=float(counts) * 1.5)
             _get_loss_ratio(output)
 
             if status != 0:
-                raise error.TestFail("Ping failed, status: %s,"
-                                     " output: %s" % (status, output))
+                test.fail("Ping returns non-zero value %s" % output)
 
-        error.context("Flood ping test", logging.info)
-        utils_test.ping(ip, None, flood=True, output_func=None,
-                        interface=interface, timeout=flood_minutes * 60)
+    if ping_ext_host:
+        error_context.context("Ping test from guest side,"
+                              " dest: '%s'" % ext_host, logging.info)
+        pkt_sizes = packet_sizes
+
+        is_windows = False
+        if params.get("os_type") == "windows":
+            # Windows doesn't support ping with packet larger than '65500'
+            pkt_sizes = [p for p in packet_sizes if int(p) < 65500]
```

Comment:
Is this line the reason why you `pkt_sizes = packet_sizes` in line131?
you can use `packet_sizes` for the place you used `pkt_sizes`, eg
`packet_sizes = [p for p in packet_sizes if int(p) < 65500]`
Based on your code, there is no reason t keep two copies of `packet_sizes`